dask_ml/preprocessing/_block_transformer.py

```diff
@@ -0,0 +1,63 @@
+import dask.array as da
+import dask.dataframe as dd
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+
+from dask_ml.utils import check_array, handle_zeros_in_scale
+
+
+class BlockTransformer(BaseEstimator, TransformerMixin):
+    """Construct a transformer from a an arbitrary callable
+
+    The BlockTransformer forwards the blocks of the X arguments to a user-defined
+    function or function-object an returns the result of this operation.
+    This is useful for stateless operations, that can be performed on the cell or
+    block level, such as taking the log of frequencies. In general the transformer
+    is not suitable for e.g. standardization tasks as this requires information for
+    a complete columns
+
+    Parameters
+    ----------
+    func : callable
+        The callable to use for the transformation.
+
+    validate : bool, optional default=True
+         Indicate that the input X array should be checked before calling
+        ``func``.
+
+    kw_args : dict, optional
+        Dictionary of additional keyword arguments to pass to func.
```

Comment:
An example here would be welcome. Perhaps a DataFrame hasher like

```python
import dask.datasets
import pandas as pd

df = dask.datasets.timeseries()
df

trn = BlockTransformer(pd.util.hash_pandas_object)
trn.transform(df)
```

Reply:
And you can show an example like `index=False` for passing through kwargs.

Reply:
good idea